src/Components/Endpoints/src/Rendering/TypeNameHash.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Components.Endpoints;
+
+// Internal for testing.
+internal class TypeNameHash
+{
+    public const int MaxStackBufferSize = 1024;
+
+    public static string Compute(Type type)
+    {
+        if (type.FullName is not { } typeName)
+        {
+            throw new InvalidOperationException($"Cannot compute a hash for a type without a {nameof(Type.FullName)}.");
+        }
+
+        Span<byte> typeNameBytes = stackalloc byte[MaxStackBufferSize];
+
+        if (!Encoding.UTF8.TryGetBytes(typeName, typeNameBytes, out var written))
+        {
+            typeNameBytes = Encoding.UTF8.GetBytes(typeName);
+            written = typeNameBytes.Length;
+        }
+
+        Span<byte> typeNameHashBytes = stackalloc byte[SHA256.HashSizeInBytes];
+        SHA256.HashData(typeNameBytes[..written], typeNameHashBytes);
+
+        return Convert.ToHexString(typeNameHashBytes);
```

Comment:
Between the SHA computation and the string allocation, is this something that could benefit from caching? Types cardinality is fixed, could be indexed on the type name.

Reply:
This is already cached [here](https://github.com/dotnet/aspnetcore/pull/52316/files#diff-8e96e4c2f12dfb979f9220c234212e5cf2ca84afd730e3115fa80c4af5e61f6dR193).